### ISSUE TITLE
Shared folders issue, trimming trailing / doesn't work?

### ISSUE THREAD

**Issue by milankyncl** [deployphp/deployer#2647]
- Deployer version: 6.8.0
- Deployment OS: Debian 10

Hey, I just created fresh Symfony project and wanted to setup Deployer to take care of deploying code for me. I just noticed weird issue in the process of creating shared folders:

I have standard Symfony 5 project structure with `var/` folder, so I just configured shared folders like this:
```php
set('shared_dirs', [
    'var/log/',
    'var/sessions',
]);
```

But when I run the deploy process, both folders are being dropped after updating code from git, so the deploy fails on creating symlinks from shared folder.
After investigating I noticed [this](https://github.com/deployphp/deployer/blob/master/recipe/deploy/shared.php#L37) can be the source of the problem. Or can you explain me why is the folder being removed before creating symlink? 

Best regards.

**Comment by Schrank** [comment]
I'm on vacation, so no testing here, but theory:
You deploy your code, deployer `rm -rf` the shared directories from your deployed code. So if you have a log or session directory on git, then it is deleted, BEFORE the shared directory is symlinked.

then we™ make sure the parent directory exists:
https://github.com/deployphp/deployer/blob/master/recipe/deploy/shared.php#L41

and then we create a symlink:
https://github.com/deployphp/deployer/blob/master/recipe/deploy/shared.php#L44

TL;DR: I don't know what your problem is, but this it isn't most likely 😅

**Comment by milankyncl** [comment]
Thanks for the fast reply of yours and sorry for disturbing your vacation 😄 

OK, that makes sense, I realised that there is no need for the target folder to exists, when I want to create symbolic link and I was able to solve this with updating the paths of shared folders:

When I try to use it like this:
```php
set('shared_dirs', [
    'var/log/',
    'var/sessions/',
]);
```
It fails with 
```
 The command "ln -nfs --relative /var/www/html/shared/var/log/ /var/www/html/releases/9/var/log/" failed.  
                                                                                                                                        
  Exit Code: 1 (General error)                                                                                                          
                                                                                                                                        
  Host Name: XXX                                                                                        
                                                                                                                                        
  ================                                                                                                                      
  ln: failed to create symbolic link '/var/www/html/releases/9/var/log/': No such file or directory
```
So it tries to create symbolink link of folder `var/log/` inside of `/var/www/html/releases/9/var/log/`, but not as `/var/www/html/releases/9/var/log/` (I hope my description is clear 😄 )

But when I use it like this:

```php
set('shared_dirs', [
    'var/log',
    'var/sessions',
]);
```

It goes pretty well. I think there is a problem with the target folder path (the path should be trimmed, without the ending "/").

**Comment by Schrank** [comment]
Time to implement a test and make sure it works :-)

**Comment by antonmedv** [comment]
Fixed
